.buildkite/scripts/common.sh

```diff
@@ -146,3 +146,24 @@ check_is_arm() {
     is_arm=""
   fi
 }
+
+git_diff() {
+  patterns=$1
+  patterns_formatted="$(echo "$patterns" | tr ':' '\n')"
+
+  if git diff --name-only HEAD@{1} HEAD | grep -qE -e "$patterns_formatted"; then
+    echo true
+  else
+    echo false
+  fi
+}
+
+are_files_changed() {
+  patterns=$1
+
+  if [[ $(git_diff "$patterns") == false ]]; then
+    echo "No files changed in $patterns"
+    exit 0;
+  fi
+}
```

Comment:
I think it would be better that this functions returns 1 or 0 , and the caller should be the one deciding if it should exit the script.

WDYT ?

```suggestion
are_files_changed() {
  patterns=$1

  if [[ $(git_diff "$patterns") == false ]]; then
    echo "No files changed in $patterns"
    return 1
  fi
  return 0
}
```